### DISCUSSION TITLE
Where do I see docs for `fade`?

### DISCUSSION THREAD

**Discussion by xrd** [ianthehenry/toodle.studio#8] [Q&A] [answered]
I've been playing more with my Hypotrochoid example (https://en.wikipedia.org/wiki/Hypotrochoid)

I have this code which is great in that it visualizes the circles used to create the Hypotrochoid. I wanted to have the circles fade out but leave the path drawn by the Hypotrochoid. I assumed if I added `fade` inside the `cloodle`  that it would only fade the lines for that cloodle, but I believe I understand `fade` now to be "clear the entire screen" rather than clear what has been drawn in this cloodle.

Is there a place to review the docs for fade? Is this something I could perhaps do inside the JS console? Do I need to read the source (which is fine, just looking for what's the most efficient and best self serve method).

And, is there a better way to do what I want to do, leave the path drawn and slowly fade out the circles as the path progresses?

```
(defn hypotrochoid [f r R t d]
  (+
    (* d (f (* t (/ (- R r) r))))
    (* (- R r) (f t))))

(def d 50)
(def r 30)
(def R 50)

(cloodle 
  (def t (/ age 100))
  (def x (hypotrochoid cos r R t d))
  (def y (hypotrochoid sin r R t (- d)))
  # (fade 0.02)
  [ x y ]
)

(cloodle 
  (def t (/ age 100))
  (def x (hypotrochoid cos r R t d))
  (def y (hypotrochoid sin r R t (- d)))
  (fade 0.02)
  [ (+ x (* r (cos age))) (+ y (* r (sin age))) ]
)
```

**Comment by ianthehenry** [comment, marked as answer]
"docs"

Indeed fade fades out the entire image, and the number just determines the rate that that happens. Fading out specific lines would be 1. really cool and 2. much more expensive, because it would require keeping a list of "lines to draw" and re-drawing them on every frame -- in the correct z-order!

There aren't really any docs for any of this, [reading the source](https://github.com/ianthehenry/toodle.studio/blob/c618378bbebee2b68231e9ce1f7b69acd5530739/ui/app.tsx#L352) is kinda the only option.

**Reply by xrd** [reply]
That's great, I'm happy (and my level of comfort is higher) to read JavaScript. Thank you.

**Reply by xrd** [reply]
But, one last question: I'm not entirely sure of all the functions to be used inside of toodle. Can I load up the code inside of a Janet REPL and use the docs inside the to at least investigate what's in scope for toodle runtime? Even seeing the arity would be interesting for example, i think that was an issue with my first attempts at playing with toodle and cloodle.

**Reply by ianthehenry** [reply]
Not exactly, because the environment that the script runs in is constructed dynamically and kinda weirdly. And also it's all completely undocumented :p

But it's basically these two files:

- https://github.com/ianthehenry/toodle.studio/blob/master/src/dsl.janet
- https://github.com/ianthehenry/toodle.studio/blob/master/src/helpers.janet

(I think just the examples serve as the most useful source of documentation, even though they are not very useful.)
